uniswap-v2/contracts/psp22/lib.rs

```diff
@@ -71,15 +71,24 @@ pub mod token {
 
     impl MyPSP22 {
         #[ink(constructor)]
-        pub fn new(total_supply: Balance) -> Self {
+        pub fn new(
+            total_supply: Balance,
+            name: Option<String>,
+            symbol: Option<String>,
+            decimals: u8,
+        ) -> Self {
             let mut instance = Self::default();
-            instance.metadata.name = Some(String::from("UNI TOKEN"));
-            instance.metadata.symbol = Some(String::from("UNI"));
-            instance.metadata.decimals = 18;
+            instance.metadata.name = name;
+            instance.metadata.symbol = symbol;
+            instance.metadata.decimals = decimals;
             instance
                 ._mint_to(instance.env().caller(), total_supply)
                 .expect("Should mint");
             instance
         }
+        #[ink(message)]
+        pub fn mint(&mut self, account: AccountId, amount: Balance) -> Result<(), PSP22Error> {
```

Comment:
Why is this needed ?

Reply:
Ah, mint actually isn't used in the integration test. Permissionless mint, however, is necessary to let user test our template deployed on Shibuya.

Reply:
Similar reason to https://github.com/AstarNetwork/wasm-showcase-dapps/pull/34#discussion_r1040711707

Reply:
Then can you add a comment that this should only be used for test purpose please

Reply:
Got it!

Reply:
Done